com.zsmartsystems.zigbee/src/main/java/com/zsmartsystems/zigbee/transaction/ZigBeeTransactionManager.java

```diff
@@ -65,7 +65,7 @@
  * attempt to send the next transaction in order to keep the transport layer full, while also fulfilling the various
  * constraints in the queues.
  * <p>
- * When sending, queues are polled in random order to ensure that all queues get a fair chance at sending data.
+ * When sending, queues are polled in order to ensure that all queues get a fair chance at sending data.
```

Comment:
I'm a little worried that these two statements can't both be true. We want to have a random release rather than sending based on some queue order - does you altered code still provide this?

I guess the `ConcurrentLinkedQueue` allows duplicate entries, so you now maintain a queue of queues, with the outer queue providing duplicates of the transaction queues. Therefore this now releases commands broadly in the order in which they are queued?

I think this doc should be updated to describe how this works as it's not super clear from a quick look at tthe code and I'd like to ensure it's documented before we merge and wonder how it works in 12 months time :)..

Reply:
Exactly. The outer queue avoids needing to reshuffle transaction queues and works even if a transaction queue is shut down. I'll rework the docs.